unit/doc.go

```diff
@@ -7,76 +7,103 @@
-// the New function and the Dimensions map. For example, the code
-//		acc := New(9.81, Dimensions{LengthDim:1, TimeDim: -2})
+//
+// 	l := 0.001 * unit.Meter
+// 	k := 1 * unit.Milli * unit.Meter
+// 	j := unit.Length(0.001)
+//
+//
+// Additional SI-derived static units can also be defined by adding types that
+// satisfy the Uniter interface described below.
+//
+// Dynamic user-extensible unit system
+//
+// The unit package also provides the Unit type, a representation of a general
+// dimensional value. Unit can be used to help prevent errors of dimensionality
+// when multiplying or dividing dimensional numbers defined a run time. New
+// variables of type Unit can be created with the New function and the
+// Dimensions map. For example, the code
+//
+// 	acc := unit.New(9.81, Dimensions{LengthDim: 1, TimeDim: -2})
+//
 // creates a variable "acc" which has a value of 9.81 m/s^2. Methods of
 // unit can be used to modify this value, for example:
-// 		acc.Mul(1.0 * unit.Kilogram).Mul(1 * unit.Meter)
+//
+// 	acc.Mul(1.0 * unit.Kilogram).Mul(1 * unit.Meter)
+//
 // To convert the unit back into a typed float64 value, the From methods
 // of the dimensional types should be used. From will return an error if the
 // dimensions do not match.
-// 		var energy unit.Energy
-//		err := (*energy).From(acc)
+//
+// 	var energy unit.Energy
+// 	err := energy.From(acc)
+//
 // Domain-specific problems may need custom dimensions, and for this purpose
 // NewDimension should be used to help avoid accidental overlap between
 // packages. For example, results from a blood test may be measured in
 // "White blood cells per slide". In this case, NewDimension should be
 // used to create a 'WhiteBloodCell' dimension. NewDimension takes in a
 // string which will be used for printing that dimension, and will return
-// a unique dimension number. NewDimension should not be
-// used, however, to create the unit of 'Slide', because in this case slide
-// is just a measurement of area. Instead, a constant could be defined.
-//		const Slide unit.Area =  0.001875 // m^2
-// Note that Unit cannot catch all errors related to dimensionality.
+// a unique dimension number.
+//
+// 	wbc := unit.NewDimension("WhiteBloodCell")
+//
+// NewDimension should not be used, however, to create the unit of 'Slide',
+// because in this case slide is just a measurement of volume. Instead, a
+// constant could be defined.
+//
+// 	const Slide unit.Volume =  0.1 * unit.Microliter
+//
+// Note that unit cannot catch all errors related to dimensionality.
 // Different physical ideas are sometimes expressed with the same dimensions
-// and Unit is incapable of catching these mismatches. For example, energy and
+// and unit is incapable of catching these mismatches. For example, energy and
 // torque are both expressed as force times distance (Newton-meters in SI),
 // but it is wrong to say that a torque of 10 N-m is the same as 10 J, even
 // though the dimensions agree.
```

Comment:
We should add something like: The arithmetic in unit cannot capture these errors, but the use of new defined units can capture these errors for compile-time safety. `type Force float64` and `type Torque float64` are different types and can be used to catch compile-time errors.

(that english is terrible, I'm sorry, but hopefully you get the gist)

Reply:
I've added a sentence to that effect and an example.

Reply:
We should probably add `Torque`. Though this breaks things unless with add an orthogonal length dimension.